settings.nix

```diff
@@ -2084,6 +2152,15 @@ with docs.lib; rec {
         (plain "environment" (mapAttrsToList leaf cfg.environment))
         (plain "binds" (mapAttrsToList bind cfg.binds))
 
+        (map workspace (sort (a: b: a.sort-name < b.sort-name)
+          (mapAttrsToList (key: cfg:
+            cfg
+            // {
+              sort-name = key;
+              name = cfg.name or key;
```

Comment:
Won't this always use `cfg.name`?  `or` will only use the rhs if the attrpath doesn't exist, and doesn't care for nulls, see e.g. in a repl ` {a = null;}.a or throw "unreachable; this evaluates to null because a is null"`

Reply:
Oh, indeed, it will. I'll adjust in a bit.

Reply:
Updated!